### PR TITLE
Fix phrasing UPE missing currencies notice text

### LINKED ISSUE #2720: Better notice when adding currencies automatically upon UPE method selection
When selecting an UPE method that requires a currency that is not presently enabled, there is a notice that stated the currency is being added. Initially this notice was for multiple methods and a single currency, however, as of https://github.com/Automattic/woocommerce-payments/pull/2652 there are now four different states:

One method with one currency: 
> Sofort require additional currencies, so we’ll add Euro (€) to your store. You can view & manage currencies later in settings.

One method with two (or more) currencies:
> Przelewy24 (P24) require additional currencies, so we’ll add Euro (€) and Polish złoty (zł) to your store. You can view & manage currencies later in settings.

Two (or more) methods with one currency:
> Sofort and giropay require additional currencies, so we’ll add Euro (€) to your store. You can view & manage currencies later in settings.

Two (or more) methods with two (or more) currencies:
>  Sofort and Przelewy24 (P24) require additional currencies, so we’ll add Euro (€) and Polish złoty (zł) to your store. You can view & manage currencies later in settings.

The wording here should be updated to handle plurals better.

### PR DESCRIPTION
**Pull request by tpaksu**
Fixes #2720

#### Changes proposed in this Pull Request

This PR fixes the phrasing of the missing currencies notice on the Add UPE methods modal. 

Previously, the notice contained:

- One method with one currency:

> Sofort require additional currencies, so we’ll add Euro (€) to your store. You can view & manage currencies later in settings.

- One method with two (or more) currencies:

> Przelewy24 (P24) require additional currencies, so we’ll add Euro (€) and Polish złoty (zł) to your store. You can view & manage currencies later in settings.

- Two (or more) methods with one currency:

> Sofort and giropay require additional currencies, so we’ll add Euro (€) to your store. You can view & manage currencies later in settings.

- Two (or more) methods with two (or more) currencies:

> Sofort and Przelewy24 (P24) require additional currencies, so we’ll add Euro (€) and Polish złoty (zł) to your store. You can view & manage currencies later in settings.

Now they contain:

> [Payment methods list] (require|requires) (an additional currency|additional currencies), so we’ll add [Currencies list] to your store. You can view & manage currencies later in settings.

And the list items are concatenated like this in the sentence:

> 1 : A
> 2 : A and B
> 3+: A, B, and C

#### Testing instructions

* Ensure you have UPE enabled in the dev tools page.
* Remove all currencies from Multi-currency page.
* Go to **Payments > Settings* page.
* Try to enable one or more UPE payment in the modal, observe the text when you select one or more providers from that list.

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

### CODE REVIEW

File: client/components/currency-information-for-methods/index.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ const ListToCommaSeparatedSentencePartConverter = ( items ) => {
 	return (
 		items.join( ', ' ) +
 		' ' +
```
Comment (jessepearson) [inline client/components/currency-information-for-methods/index.js:26]: This line isn't needed, it adds an extra space before the `, and`.

**Review by jessepearson — APPROVED** [review]
Pre-approving pending removing the line mentioned. Thanks for taking care of this 😁

File: client/components/currency-information-for-methods/index.js (hunk 1 above)
Comment (tpaksu) [reply client/components/currency-information-for-methods/index.js]: Removed it, thanks for the heads-up!

**Review by jessepearson — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**